pkgs/applications/version-management/rcs/default.nix

```diff
@@ -1,36 +1,17 @@
 { lib, stdenv, fetchurl, fetchpatch, ed }:
 
 stdenv.mkDerivation rec {
-  name = "rcs-5.9.4";
+  pname = "rcs";
+  version = "5.10.0";
 
   src = fetchurl {
-    url = "mirror://gnu/rcs/${name}.tar.xz";
-    sha256 = "1zsx7bb0rgvvvisiy4zlixf56ay8wbd9qqqcp1a1g0m1gl6mlg86";
+    url = "mirror://gnu/rcs/${pname}-${version}.tar.xz";
+    sha256 = "sha256-Og2flYx60wPkdehjRlSXTtvm3rOkVEkfOFfcGIm6xcU";
   };
 
-  buildInputs = [ ed ];
+  nativeBuildInputs = [ ed ];
```

Comment:
This isn't quite right.  The configure script wants to find `ed` so that it can embed its path in the binary.  Putting it in `nativeBuildInputs` will mean it finds the build system's ed, so the package won't work when cross-compiled.  We actually have a similar (but invisible) problem with diffutils, which it autodetects from stdenv even though we're not setting it as any sort of build input in this expression.

I suggest the following fix, which sets the path to all the required runtime programs manually, rather than searching PATH, and adds a `disallowedReferences` check to ensure they can't accidentally come back in future:

```diff
diff --git i/pkgs/applications/version-management/rcs/default.nix w/pkgs/applications/version-management/rcs/default.nix
index 0c76e520dd8..7e97049489f 100644
--- i/pkgs/applications/version-management/rcs/default.nix
+++ w/pkgs/applications/version-management/rcs/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl, fetchpatch, ed }:
+{ lib, stdenv, fetchurl, fetchpatch, buildPackages, diffutils, ed }:
 
 stdenv.mkDerivation rec {
   pname = "rcs";
@@ -9,7 +9,13 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-Og2flYx60wPkdehjRlSXTtvm3rOkVEkfOFfcGIm6xcU";
   };
 
-  nativeBuildInputs = [ ed ];
+  ac_cv_path_ED = "${ed}/bin/ed";
+  DIFF = "${diffutils}/bin/diff";
+  DIFF3 = "${diffutils}/bin/diff3";
+
+  disallowedReferences =
+    lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform)
+      [ buildPackages.diffutils buildPackages.ed ];
 
   doCheck = !stdenv.isDarwin;
 

```